### DISCUSSION TITLE
Anyone else have a good way to rename/add field keys and values after lots of notes for a particular fileClass already exist?

### DISCUSSION THREAD

**Discussion by oakla** [mdelobelle/metadatamenu#713] [General]
I had a fileClass with some existing notes for which I wanted to redesign the fields. For example I wanted to add some new properties and rename some existing values (for example, give them a numbered prefix so they would sort more easily in Dataview queries). I couldn't work out how to do it with Metadata Menu, so I wrote a Python script to do it instead, and manually updated the Metadata Menu field settings for the fileClass.

It looked fine once the script was done, but if I try to use Metadata Menu to update the changed notes, it kind of ignores the frontmatter that's already in there and either creates a new frontmatter or adds the field as an extra inline field (and just leaves the existing field in the frontmatter).

I raised an issue about this (#712), but I was wondering if anyone else has similar experiences, solutions or advice?

**Comment by LynetteCullens** [comment]
You could start here. https://github.com/mdelobelle/metadatamenu/discussions/591#discussion-6232570
